contrib/openssl-cmake/CMakeLists.txt

```diff
@@ -1447,6 +1415,11 @@ set(SSL_SRC
     ${OPENSSL_SOURCE_DIR}/ssl/statem/statem_srvr.c
 )
 
+# Disable all deprecated API
+add_definitions(-DOPENSSL_NO_DEPRECATED)
+add_definitions(-DOPENSSL_NO_ENGINE)
```

Comment:
Is `no-engine` really deprecated (l. 1418)? What kind of functionality does it disallow and why don't we want it?

(let's add a comment)

Reply:
https://docs.openssl.org/3.3/man3/ENGINE_add/#history